dask/array/tests/test_cupy.py

```diff
@@ -815,6 +815,20 @@ def test_sfqr(m, n, chunks, error_type):
             q, r = da.linalg.sfqr(data)
 
 
+def test_sparse_hstack_vstack_csr():
+    pytest.importorskip("cupyx")
+    x = cupy.array([2.0, 1.0, 5.0, 2.0, 1.0], dtype=cupy.float32)
+    x = cupy.expand_dims(x, axis=0)
+
+    sp = da.from_array(x, chunks=(1000), asarray=False, fancy=False)
```

Comment:
Can we try a chunk size smaller than the array? Not sure if the concatenate method is used in this case as it is just one big chunk. Would suggest maybe `2` or `3`?

Reply:
Inline with the change to make the matrix 2-D above, this could be reasonable chunking to ensure concatenation along both dimensions is tested.

```suggestion
    sp = da.from_array(x, chunks=(2, 3), asarray=False, fancy=False)
```